repos/system_upgrade/common/actors/targetuserspacecreator/tests/unit_test_targetuserspacecreator.py

```diff
@@ -1325,3 +1329,60 @@ def has_dbgmsg(substr):
     )
     assert has_dbgmsg('SKIP the tls/certs/server.crt file: not owned by any rpm')
     assert has_dbgmsg('Found the file owned by an rpm: rpm-gpg/RPM-GPG-KEY-2.')
+
+
+def test_writing_stream_varfile(monkeypatch):
+
+    monkeypatch.setattr(userspacegen.api, 'current_actor', CurrentActorMocked())
+    monkeypatch.setattr(userspacegen, 'get_target_major_version', lambda: '10')
+
+    with tempfile.NamedTemporaryFile(mode='w+') as tmpf:
+        tmpf.write('incorrect-stream-value\n')
+        tmpf.flush()
+        userspacegen.adjust_dnf_stream_variable(MockedMountingBase, tmpf.name)
+        tmpf.seek(0)
+        content = tmpf.read()
+
+    assert content == '10-stream\n'
+
+
+def test_failing_stream_varfile_write(monkeypatch):
+    monkeypatch.setattr(userspacegen.api, 'current_actor', CurrentActorMocked())
+    monkeypatch.setattr(userspacegen, 'get_target_major_version', lambda: '10')
+    with pytest.raises(StopActorExecutionError) as err:
+        userspacegen.adjust_dnf_stream_variable(MockedMountingBase, '/path/not/exists')
+
+    assert 'Failed to adjust dnf variable' in str(err.value)
+
+
+@pytest.mark.parametrize("distro,should_adjust", [('rhel', False), ('centos', True)])
+def test_if_adjust_dnf_stream_variable_only_for_centos(monkeypatch, distro, should_adjust):
+
+    def mock_set_container_mode(MockedMountingBase):
+        pass
+
+    def mock_switch_certificate(MockedMountingBase, testInData, cert_path=None):
+        pass
+
+    def mock__install_custom_repofiles(MockedMountingBase, testInData):
+        pass
+
+    def mock_adjust_stream_variable(context, varfile='/etc/dnf/vars/stream'):
+        nonlocal adjust_called
+        adjust_called = True
+
+    def mock_gather_target_repositories(MockedMountingBase, testInData):
+        pass
+
+    monkeypatch.setattr(userspacegen.api, 'current_actor', CurrentActorMocked(release_id=distro))
+    monkeypatch.setattr(userspacegen, 'get_target_major_version', lambda: '10')
+    monkeypatch.setattr(rhsm, 'set_container_mode', mock_set_container_mode)
+    monkeypatch.setattr(rhsm, 'switch_certificate', mock_switch_certificate)
+    monkeypatch.setattr(userspacegen, '_install_custom_repofiles', mock__install_custom_repofiles)
+    monkeypatch.setattr(userspacegen, 'adjust_dnf_stream_variable', mock_adjust_stream_variable)
+    monkeypatch.setattr(userspacegen, 'gather_target_repositories', mock_gather_target_repositories)
```

Comment:
wrt the `pass` functions, they are alright, though you can make this a lot shorter using a single function. If you want to keep it at it is, I would at least ask to use the real parameter names instead of `testInData`.

wrt the `mock_adjust_stream_variable`, I would assert whether the varfile is indeed /etc/dnf/vars/stream.

```suggestion
    def do_nothing(*args, **kwargs):
        pass

    def mock_adjust_stream_variable(context, varfile='/etc/dnf/vars/stream'):
        assert varfile = '/etc/dnf/vars/stream'
        nonlocal adjust_called
        adjust_called = True

    monkeypatch.setattr(userspacegen.api, 'current_actor', CurrentActorMocked(release_id=distro))
    monkeypatch.setattr(userspacegen, 'get_target_major_version', lambda: '10')
    monkeypatch.setattr(rhsm, 'set_container_mode', do_nothing)
    monkeypatch.setattr(rhsm, 'switch_certificate', do_nothing)
    monkeypatch.setattr(userspacegen, '_install_custom_repofiles', do_nothing)
    monkeypatch.setattr(userspacegen, 'adjust_dnf_stream_variable', mock_adjust_stream_variable)
    monkeypatch.setattr(userspacegen, 'gather_target_repositories', do_nothing)
```

Reply:
I removed assertion  `assert varfile = '/etc/dnf/vars/stream'` from mock_adjust_stream_variable because I make this value default. I'm not sure if assert is needed here, however it doesn't harm

Reply:
Yeah, right now it shouldn't fail unless a non-default incorrect argument is passed, but in the future the code may get refactored. It doesn't do any harm to keep there.